pkg/orchestration/k8s/cluster.go

```diff
@@ -233,6 +233,22 @@ func (c *Cluster) FullNodeNames() (names []string) {
 	return
 }
 
+// ShuffledFullNodeClients returns a list of full node clients shuffled
+func (c *Cluster) ShuffledFullNodeClients(ctx context.Context, r *rand.Rand) ([]*bee.Client, error) {
+	cls, err := c.NodesClients(ctx)
```

Comment:
This will return all nodes, not only full nodes, so maybe check with implementation of `FullNodeNames()` method and combine it to extract only full nodes.